nptl/allocatestack.c

```diff
@@ -367,7 +367,12 @@ allocate_stack (const struct pthread_attr *attr, struct pthread **pdp,
 		// 	MAP_PRIVATE | MAP_ANONYMOUS | MAP_STACK, -1, 0);
 
     // Dennis Edit: Replacement mmap with malloc
-    size = 65664;
+    // Qianxi Edit: changed the malloc size to 16416 (1/4 of the original)
+    //              since currently there is an issue on malloc that it cannot
+    //              automatically grow the memory if memory is running out.
+    //              Once the issue is fixed, we might be able to change the size
+    //              back
+    size = 16416;
```

Comment:
what is this size?

Reply:
This is the thread stack size. I'll add a note here later

Reply:
Previously it is using mmap to request for stack space and we replaced it with malloc. I think once mmap stuff is finished, this could be changed back and the size issue would also go away.